src/platform/mt8186/platform.c

```diff
@@ -0,0 +1,215 @@
+#include <ipc/dai.h>
+#include <ipc/header.h>
+#include <ipc/info.h>
+#include <kernel/abi.h>
+#include <kernel/ext_manifest.h>
+#include <sof_versions.h>
+#include <errno.h>
+#include <stdint.h>
+#include <xtensa/hal.h>
+#include <platform/drivers/timer.h>
+
+struct sof;
+
+static const struct sof_ipc_fw_ready ready
+	__section(".fw_ready") = {
+	.hdr = {
+		.cmd = SOF_IPC_FW_READY,
+		.size = sizeof(struct sof_ipc_fw_ready),
+	},
+	/* dspbox is for DSP initiated IPC, hostbox is for host initiated IPC */
+	.version = {
+		.hdr.size = sizeof(struct sof_ipc_fw_version),
+		.micro = SOF_MICRO,
+		.minor = SOF_MINOR,
+		.major = SOF_MAJOR,
+		.tag = SOF_TAG,
+		.abi_version = SOF_ABI_VERSION,
+		.src_hash = SOF_SRC_HASH,
+	},
+	.flags = DEBUG_SET_FW_READY_FLAGS,
+};
+
+#define NUM_MTK_WINDOWS 6
+
+const struct ext_man_windows xsram_window
+		__aligned(EXT_MAN_ALIGN) __section(".fw_metadata") __unused = {
+	.hdr = {
+		.type = EXT_MAN_ELEM_WINDOW,
+		.elem_size = ALIGN_UP_COMPILE(sizeof(struct ext_man_windows), EXT_MAN_ALIGN),
+	},
+	.window = {
+		.ext_hdr	= {
+			.hdr.cmd = SOF_IPC_FW_READY,
+			.hdr.size = sizeof(struct sof_ipc_window),
+			.type	= SOF_IPC_EXT_WINDOW,
+		},
+		.num_windows	= NUM_MTK_WINDOWS,
+		.window	= {
+			{
+				.type	= SOF_IPC_REGION_UPBOX,
+				.id	= 0,	/* map to host window 0 */
+				.flags	= 0,
+				.size	= MAILBOX_DSPBOX_SIZE,
+				.offset	= MAILBOX_DSPBOX_OFFSET,
+			},
+			{
+				.type	= SOF_IPC_REGION_DOWNBOX,
+				.id	= 0,	/* map to host window 0 */
+				.flags	= 0,
+				.size	= MAILBOX_HOSTBOX_SIZE,
+				.offset	= MAILBOX_HOSTBOX_OFFSET,
+			},
+			{
+				.type	= SOF_IPC_REGION_DEBUG,
+				.id	= 0,	/* map to host window 0 */
+				.flags	= 0,
+				.size	= MAILBOX_DEBUG_SIZE,
+				.offset	= MAILBOX_DEBUG_OFFSET,
+			},
+			{
+				.type	= SOF_IPC_REGION_TRACE,
+				.id	= 0,	/* map to host window 0 */
+				.flags	= 0,
+				.size	= MAILBOX_TRACE_SIZE,
+				.offset	= MAILBOX_TRACE_OFFSET,
+			},
+			{
+				.type	= SOF_IPC_REGION_STREAM,
+				.id	= 0,	/* map to host window 0 */
+				.flags	= 0,
+				.size	= MAILBOX_STREAM_SIZE,
+				.offset	= MAILBOX_STREAM_OFFSET,
+			},
+			{
+				.type	= SOF_IPC_REGION_EXCEPTION,
+				.id	= 0,	/* map to host window 0 */
+				.flags	= 0,
+				.size	= MAILBOX_EXCEPTION_SIZE,
+				.offset	= MAILBOX_EXCEPTION_OFFSET,
+			},
+		},
+	}
+};
+
+static SHARED_DATA struct timer timer = {
+	.id = OSTIMER0,
+	.irq = MTK_DSP_IRQ_OSTIMER32,
+};
+
+/* Override the default MPU setup. This table matches the memory map
+ * of the 'sample_controller' core and will need to be modified for
+ * other cores.
+ * NOTE: This table sets up all of external memory as shared uncached.
+ * For best results, edit the LSP memory map to create a separate
+ * section in shared memory, place all sections that need to be uncached
+ * into that section, and only map that section uncached. See README
+ * for more details.
+ */
+const unsigned int __xt_mpu_init_table_size __attribute__((section(".ResetVector.text"))) = 5;
```

Comment:
should be 6 , I think it also can use #define ARRAYSIZE(a) (sizeof(a) / sizeof(*(a))) for it?